src/games/rcll/robots.clp

```diff
@@ -99,6 +99,14 @@
 	(printout t "Robot " ?robot:number " scheduled for maintenance cycle " ?cycle crlf)
 	(modify ?robot (state MAINTENANCE) (maintenance-start-time ?ctime)
 		(maintenance-cycles ?cycle) (maintenance-warning-sent FALSE))
+	(bind ?cycle-cost (nth$ ?cycle ?*MAINTENANCE-COST*))
+	(if (neq ?cycle-cost nil)
+	 then
+		(assert (points (game-time ?game-time)
+		                (points (* -1 ?cycle-cost))
+		                (team ?robot:team-color) (phase PRODUCTION)
```

Comment:
It is also possible to declare maintenance during exploration phase. Therefore points have to be deducted in the correct phase

Reply:
Good catch! I pushed a fix